src/bls_ops.rs

```diff
@@ -172,14 +182,19 @@ pub fn op_bls_g2_multiply(a: &mut Allocator, input: NodePtr, max_cost: Cost) ->
     new_atom_and_cost(a, cost, &total.to_compressed())
 }
 
-pub fn op_bls_g2_negate(a: &mut Allocator, input: NodePtr, max_cost: Cost) -> Response {
+pub fn op_bls_g2_negate(a: &mut Allocator, input: NodePtr, _max_cost: Cost) -> Response {
     let args = Node::new(a, input);
     check_arg_count(&args, 1, "g2_negate")?;
-    let cost = BLS_G2_NEGATE_BASE_COST;
-    check_cost(a, cost, max_cost)?;
-    let point = g2_atom(args.first()?)?;
-    let total = -point;
-    new_atom_and_cost(a, cost, &total.to_compressed())
+
+    // we don't validate the point. We may want to soft fork-in validating the
+    // point once the allocator preserves native representation of points
+    let blob = atom(args.first()?, "G2 atom")?;
+    if blob.len() != 96 {
+        return args.first()?.err("atom is not G2 size, 96 bytes");
+    }
+    let mut blob: [u8; 96] = blob.try_into().unwrap();
```

Comment:
Not a big deal, but you could just make the error case of `try_into` lead into the error case in 193

Reply:
hm. I didn't actually try, I just assumed I would get in trouble for the error types being incompatible